packages/@react-spectrum/toast/src/ToastContainer.tsx

```diff
@@ -14,19 +14,32 @@ import {classNames} from '@react-spectrum/utils';
 import React, {ReactElement} from 'react';
 import {Toast} from './';
 import toastContainerStyles from './toastContainer.css';
-import {ToastStateBase} from '@react-types/toast';
+import {ToastState} from '@react-types/toast';
+import {useProvider} from '@react-spectrum/provider';
 
-interface ToastProviderProps {
-  toasts?: ToastStateBase[],
-}
+export function ToastContainer(props: ToastState): ReactElement {
+  let {
+    onRemove,
+    toasts
+  } = props;
+  let providerProps = useProvider();
+  let toastPlacement = providerProps && providerProps.toastPlacement && providerProps.toastPlacement.split(' ');
+  let containerPosition = toastPlacement && toastPlacement[0];
+  let containerPlacement = toastPlacement && toastPlacement[1];
 
-export function ToastContainer(props: ToastProviderProps): ReactElement {
-  let renderToasts = () => props.toasts.map((toast) =>
-    (<Toast {...toast.props}>{toast.content}</Toast>)
+  let renderToasts = () => toasts.map((toast) =>
+    (<Toast key={toast.props.idKey} {...toast.props} onRemove={onRemove}>{toast.content}</Toast>)
```

Comment:
Should `{...toast.props}` be put before `key={toast.props.idKey}` to guard against possible overwrites of the key props? Maybe a non-issue since users won't use ToastContainer directly I assume and since toast props shouldn't contain a `key` prop anyways.